src/test/java/io/cdap/delta/bigquery/BigQueryUtilsTest.java

```diff
@@ -17,50 +17,315 @@
+      PowerMockito.doReturn(1L, 2L, 3L, 4L)
+        .when(BigQueryUtils.class, "executeAggregateQuery",
+              ArgumentMatchers.eq(bigQueryMock), ArgumentMatchers.any(), ArgumentMatchers.any());
+    }
+
+    @Test
+    public void testNormalizeDataSetOrTableName() {
+      // only contains number and letter
+      assertEquals("a2fs", BigQueryUtils.normalizeDatasetOrTableName("a2fs"));
+      // only contains number and letter start with number
+      assertEquals("2fas", BigQueryUtils.normalizeDatasetOrTableName("2fas"));
+      // only contains number and letter and length is 1024
+      String name = Strings.repeat("a1", 512);
+      assertEquals(name, BigQueryUtils.normalizeDatasetOrTableName(name));
+      // only contains number and letter and length is 1026
+      name = Strings.repeat("a1", 513);
+      assertEquals(name.substring(0, 1024), BigQueryUtils.normalizeDatasetOrTableName(name));
+      // contains invalid character
+      assertEquals("ab_c", BigQueryUtils.normalizeDatasetOrTableName("ab?/c"));
+      // contains space
+      assertEquals("a2_fs", BigQueryUtils.normalizeFieldName("a2 fs"));
+
+    }
+
+    @Test
+    public void testNormalizeFieldName() {
+      // only contains number and letter
+      assertEquals("a2fs", BigQueryUtils.normalizeFieldName("a2fs"));
+      // only contains number and letter start with number
+      assertEquals("_2fas", BigQueryUtils.normalizeFieldName("2fas"));
+      // only contains number and letter and length is 128
+      String name = Strings.repeat("a1", 64);
+      assertEquals(name, BigQueryUtils.normalizeFieldName(name));
+      // only contains number and letter, starts with number and length is 128
+      name = Strings.repeat("1a", 64);
+      assertEquals("_" + name.substring(0, 127), BigQueryUtils.normalizeFieldName(name));
+      // only contains number and letter and length is 130
+      name = Strings.repeat("a1", 65);
+      assertEquals(name.substring(0, 128), BigQueryUtils.normalizeFieldName(name));
+      // contains invalid character
+      assertEquals("ab_c", BigQueryUtils.normalizeFieldName("ab?/c"));
+      // contains space
+      assertEquals("a2_fs", BigQueryUtils.normalizeFieldName("a2 fs"));
+    }
+
+
+    @Test
+    public void testGetMaximumExistingSequenceNumberZeroInvocations() throws Exception {
+      // Zero Tables
+      Set<SourceTable> allTables = generateSourceTableSet(0);
+      long tableResult0 = BigQueryUtils.getMaximumExistingSequenceNumber(allTables, "testproject",
+                                                                         null, bigQueryMock, null, 1000);
+      assertEquals(0L, tableResult0);
+      PowerMockito.verifyPrivate(BigQueryUtils.class, times(0))
+        .invoke("executeAggregateQuery",
+                ArgumentMatchers.eq(bigQueryMock), ArgumentMatchers.any(), ArgumentMatchers.any());
+
+    }
+
+    @Test
+    public void testGetMaximumExistingSequenceNumberSingleInvocations() throws Exception {
```

Comment:
We dont need this test now right? or anything related to PowerMockito, since we will directly executing test against bq?

Reply:
I was thinking these are required because : 
i. These tests will run at least, incase GCP service role is not provided
ii. These tests actually checks how many invocations/calls will go to Big query depending on size of table list